### PR TITLE
Better participants sorting in sidebar tab

### PR DESCRIPTION
**Pull request by nikola-gladovic**
Signed-off-by: Nikola <nikola.gladovic@nextcloud.com>

### CODE REVIEW

**Comment by nickvergessen** [comment]
You are modifying the right sidebar. While I think it also makes sense to sort participants there, the ticket was especially about the callview, but not sure that is task that is easy. Biggest issue is that it needs lot of work in the non-vue-d webrtc lib parts and collections.

File: src/components/RightSidebar/Participants/CurrentParticipants/CurrentParticipants.vue [hunk 1]
```diff
@@ -203,6 +203,14 @@ export default {
 				return moderator1 ? -1 : 1
 			}
 
+			if (p1inCall) {
+				return participant1.attendeePermissions < participant2.attendeePermissions ? 1 : -1
+			}
+
+			if (p1inCall && participant1.attendeePermissions === participant2.attendeePermissions) {
+				return participant1.inCall < participant2.inCall ? 1 : -1
+			}
```
Comment (nickvergessen) [inline src/components/RightSidebar/Participants/CurrentParticipants/CurrentParticipants.vue:212]: The inCall attribute is already sorted by in line ~179

I think I would do it all the time (independent from inCall or not)

```suggestion
			if (participant1.attendeePermissions !== participant2.attendeePermissions) {
				return participant1.attendeePermissions < participant2.attendeePermissions ? 1 : -1
			}
```

 but only for moderators (other users dont see permissions):
* So include the participant mixin
* check with code like this if the current user is a moderator:
https://github.com/nextcloud/spreed/blob/b06c7f24886e6773219aa60d93e27ee1bb9f59d8/src/components/RightSidebar/Participants/ParticipantsList/Participant/Participant.vue#L614-L616

Comment (nikola-gladovic) [reply src/components/RightSidebar/Participants/CurrentParticipants/CurrentParticipants.vue:212]: I think already edit in the correct place (right sidebar), but I need to update new logic for admin view, other guys will not change. right?

Comment (nickvergessen) [reply src/components/RightSidebar/Participants/CurrentParticipants/CurrentParticipants.vue:212]: > but I need to update new logic for admin view, other guys will not change. right?

yes

**Comment by nikola-gladovic** [comment]
I updated this PR :)

**Review by marcoambrosini — COMMENTED** [review]
@nikola-gladovic as Joas commented, this doesn't affect the call-view, so I removed that from the title and unlinked the issue from the pr. Could you please describe what your pr does and what needs to be tested? 

Btw I'm having guests before regular users in the sidebar tab, is that intended?

Screenshot 2022-02-25 at 12 10 36

**Comment by nickvergessen** [comment]
> Btw I'm having guests before regular users in the sidebar tab, is that intended?

Also happens on master. seems it was dropped somewhere else by accident

**Comment by nickvergessen** [comment]
Moved to new issue in https://github.com/nextcloud/spreed/issues/6930

**Comment by nikola-gladovic** [comment]
Ok, thanks.
So, it will be related with #6930, not a #6697 . right?

**Comment by nickvergessen** [comment]
> So, it will be related with https://github.com/nextcloud/spreed/issues/6930, not a https://github.com/nextcloud/spreed/issues/6697 . right?

yes

**Comment by nikola-gladovic** [comment]
Done

**Pull request merged**